test/testaudioinfo.c

```diff
@@ -68,6 +70,34 @@ main(int argc, char **argv)
 
     print_devices(0);
     print_devices(1);
+    
+    
+    if (SDL_GetDefaultAudioInfo(&deviceName, &spec, 0)) {
+        SDL_Log("Error when calling SDL_GetDefaultAudioInfo: %s\n", SDL_GetError());
+    } else {
+        SDL_Log("Default audio output info found!\n");
+#if MACOSX_COREAUDIO // Device name is unavailable for iOS/tvOS
+        SDL_Log("Name: %s\n", deviceName);
+        SDL_free(deviceName);
+#endif
+        SDL_Log("Sampling Rate: %d\n", spec.freq);
+        SDL_Log("Number of Channels: %d\n", spec.channels);
+        SDL_Log("Audio Format: %d\n", spec.format);
+    }
+    
+    if (SDL_GetDefaultAudioInfo(&deviceName, &spec, 1)) {
+        SDL_Log("Error when calling SDL_GetDefaultAudioInfo: %s\n", SDL_GetError());
+    } else {
+        SDL_Log("Default audio capture info found!\n");
+#if MACOSX_COREAUDIO // Device name is unavailable for iOS/tvOS
```

Comment:
(same as above)

Reply:
Let me fix that (it wasn't executing on iOS/tvOS/iPadOS, but I have a better way to do that now)

Reply:
> Let me fix that (it wasn't executing on iOS/tvOS/iPadOS, but I have a better way to do that now)

I suggest this patch: [patch.txt](https://github.com/libsdl-org/SDL/files/9672460/patch.txt) (also inlined below.)
```diff
diff --git a/test/testaudioinfo.c b/test/testaudioinfo.c
index ee22838..c0333b3 100644
--- a/test/testaudioinfo.c
+++ b/test/testaudioinfo.c
@@ -70,35 +70,28 @@ main(int argc, char **argv)
 
     print_devices(0);
     print_devices(1);
-    
-    
+
     if (SDL_GetDefaultAudioInfo(&deviceName, &spec, 0)) {
         SDL_Log("Error when calling SDL_GetDefaultAudioInfo: %s\n", SDL_GetError());
     } else {
         SDL_Log("Default audio output info found!\n");
-#if MACOSX_COREAUDIO // Device name is unavailable for iOS/tvOS
-        SDL_Log("Name: %s\n", deviceName);
+        SDL_Log("Name: %s\n", deviceName != NULL ? deviceName : "unknown");
         SDL_free(deviceName);
-#endif
         SDL_Log("Sampling Rate: %d\n", spec.freq);
         SDL_Log("Number of Channels: %d\n", spec.channels);
         SDL_Log("Audio Format: %d\n", spec.format);
     }
-    
     if (SDL_GetDefaultAudioInfo(&deviceName, &spec, 1)) {
         SDL_Log("Error when calling SDL_GetDefaultAudioInfo: %s\n", SDL_GetError());
     } else {
         SDL_Log("Default audio capture info found!\n");
-#if MACOSX_COREAUDIO // Device name is unavailable for iOS/tvOS
-        SDL_Log("Name: %s\n", deviceName);
+        SDL_Log("Name: %s\n", deviceName != NULL ? deviceName : "unknown");
         SDL_free(deviceName);
-#endif
         SDL_Log("Sampling Rate: %d\n", spec.freq);
         SDL_Log("Number of Channels: %d\n", spec.channels);
         SDL_Log("Audio Format: %d\n", spec.format);
     }
 
-
     SDL_Quit();
     return 0;
 }
diff --git a/test/testsurround.c b/test/testsurround.c
index ff55af7..34b0f6c 100644
--- a/test/testsurround.c
+++ b/test/testsurround.c
@@ -205,18 +205,17 @@ main(int argc, char *argv[])
         SDL_Log("Error when calling SDL_GetDefaultAudioInfo: %s\n", SDL_GetError());
     } else {
         SDL_Log("Default audio output info found!\n");
-        SDL_Log("Name: %s\n", deviceName);
+        SDL_Log("Name: %s\n", deviceName != NULL ? deviceName : "unknown");
         SDL_Log("Sampling Rate: %d\n", spec.freq);
         SDL_Log("Number of Channels: %d\n", spec.channels);
         SDL_Log("Audio Format: %d\n", spec.format);
         SDL_free(deviceName);
     }
-    
     if (SDL_GetDefaultAudioInfo(&deviceName, &spec, 1)) {
         SDL_Log("Error when calling SDL_GetDefaultAudioInfo: %s\n", SDL_GetError());
     } else {
         SDL_Log("Default audio capture info found!\n");
-        SDL_Log("Name: %s\n", deviceName);
+        SDL_Log("Name: %s\n", deviceName != NULL ? deviceName : "unknown");
         SDL_Log("Sampling Rate: %d\n", spec.freq);
         SDL_Log("Number of Channels: %d\n", spec.channels);
         SDL_Log("Audio Format: %d\n", spec.format);
```